sdk/core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java

```diff
@@ -134,37 +135,76 @@ public void error(String format, Object... args) {
      * @throws NullPointerException If {@code runtimeException} is {@code null}.
      */
     public RuntimeException logExceptionAsWarning(RuntimeException runtimeException) {
         Objects.requireNonNull(runtimeException, "'runtimeException' cannot be null.");
+
+        return logCheckedExceptionAsWarning(runtimeException);
+    }
+
+    /**
+     * Logs the {@link Throwable} at the warning level and returns it to be thrown.
+     * <p>
+     * This API covers the cases where a checked exception type needs to be thrown and logged. If a {@link
+     * RuntimeException} is being logged use {@link #logExceptionAsWarning(RuntimeException)} instead.
+     *
+     * @param checkedException Throwable to be logged and returned.
+     * @param <T> Type of the Throwable being logged.
+     * @return The passed {@link Throwable}.
+     * @throws NullPointerException If {@code checkedException} is {@code null}.
+     */
+    public <T extends Throwable> T logCheckedExceptionAsWarning(T checkedException) {
+        Objects.requireNonNull(checkedException, "'checkedException' cannot be null.");
         if (!logger.isWarnEnabled()) {
-            return runtimeException;
+            return checkedException;
         }
 
-        performLogging(LogLevel.WARNING, true, runtimeException.getMessage(), runtimeException);
-        return runtimeException;
+        performLogging(LogLevel.WARNING, true, checkedException.getMessage(), checkedException);
+        return checkedException;
     }
 
     /**
      * Logs the {@link RuntimeException} at the error level and returns it to be thrown.
+     * <p>
+     * This API covers the cases where a runtime exception type needs to be thrown and logged. If a {@link Throwable} is
+     * being logged use {@link #logCheckedExceptionAsError(Throwable)} instead.
      *
      * @param runtimeException RuntimeException to be logged and returned.
      * @return The passed {@code RuntimeException}.
      * @throws NullPointerException If {@code runtimeException} is {@code null}.
      */
     public RuntimeException logExceptionAsError(RuntimeException runtimeException) {
         Objects.requireNonNull(runtimeException, "'runtimeException' cannot be null.");
+
+        return logCheckedExceptionAsError(runtimeException);
+    }
+
+    /**
+     * Logs the {@link Throwable} at the error level and returns it to be thrown.
+     * <p>
+     * This API covers the cases where a checked exception type needs to be thrown and logged. If a {@link
+     * RuntimeException} is being logged use {@link #logExceptionAsError(RuntimeException)} instead.
+     *
+     * @param checkedException Throwable to be logged and returned.
+     * @param <T> Type of the Throwable being logged.
+     * @return The passed {@link Throwable}.
+     * @throws NullPointerException If {@code checkedException} is {@code null}.
+     */
+    public <T extends Throwable> T logCheckedExceptionAsError(T checkedException) {
```

Comment:
Since this applies to all throwables, we can name this `logThrowableAsError`

Reply:
Renamed